selfhost/typechecker.jakt

```diff
@@ -1945,11 +1939,60 @@ struct Typechecker {
                             span: imported_name.span
                         )
                     }
-                }
             }
         }
+    }
 
-        return true
+    function typecheck_extern_import(mut this, import_: ParsedExternImport, scope_id: ScopeId, mut parent_namespace: ParsedNamespace) throws {
+        for f in import_.assigned_namespace.functions.iterator() {
+            if not f.linkage is External {
+                .error("Expected all functions in an `import extern` to be be external", f.name_span)
+            }
+
+            if import_.is_c and not f.generic_parameters.is_empty() {
+                .error_with_hint(
+                    format("imported function '{}' is declared to have C linkage, but is generic",f.name) 
+                    f.name_span
+                    "this function may not be generic"
+                    f.name_span)
+            }
+
+            if not f.block.stmts.is_empty() {
+                .error("imported extern function is not allowed to have a body"
+                        f.name_span)
+            }
+
+        }       
+
+        for record in import_.assigned_namespace.records.iterator() {
+            if not record.definition_linkage is External {
+                .error("Expected all records in an `import extern` to be external"
+                        record.name_span)
+            }
+            if import_.is_c and not record.generic_parameters.is_empty() {
+                .error_with_hint(
+                    format("imported {} '{}' is declared to have C linkage, but is generic", 
+                        record.record_type.record_type_name() 
+                        record.name)
+                    record.name_span, 
+                    format("this {} may not be generic", record.record_type.record_type_name())
+                    record.name_span)
+            }
+        }
+
+
+
+        parent_namespace.namespaces.push(import_.assigned_namespace)
```

Comment:
`parent_namespace` is a copy, but it copies the array reference for namespaces. I'll change the parameter since previously I was trying to  do more printing with `parent_namespace`

Reply:
Changed it to passing the reference to the parent's children array. Tried running `typecheck_predecl` there but does not seem to be recognized by the rest of the typechecker.